src/account_multisig.ts

```diff
@@ -159,54 +159,108 @@ export class Account2FA extends AccountMultisig {
         ]
         if ((await this.state()).code_hash === '11111111111111111111111111111111') {
+            const newArgs = Buffer.from(JSON.stringify({ 'num_confirmations': 2 }));
             actions.push(functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT),)
         }
         console.log('deploying multisig contract for', accountId)
-        return await super.signAndSendTransactionWithAccount(accountId, actions);
+        let result
+        try {
+            result = await this.signAndSendTransactionWithAccount(accountId, actions)
+        } catch (e) {
+            if (!/RetriesExceeded/.test(e.toString())) {
+                throw e
+            }
+            console.warn(e)
+        }
+        return await this.hasPublicKey(newLocalPublicKey) && result
     }
 
-    async disable(contractBytes: Uint8Array) {
+    async disable(contractBytes: Uint8Array, newLocalPublicKey: PublicKey) {
         const { accountId } = this
-        const accessKeys = await this.getAccessKeys()
-        const lak2fak = accessKeys.filter(({ access_key }) => 
+        let [
+            confirmOnlyKey,
+            localPublicKey,
+            accessKeys
+        ] = await Promise.all([
+            this.postSignedJson('/2fa/getAccessKey', { accountId }),
+            this.connection.signer.getPublicKey(accountId, this.connection.networkId),
+            this.getAccessKeys()
+        ])
+        confirmOnlyKey = PublicKey.from(confirmOnlyKey.publicKey)
+        const localPublicKeyStr = localPublicKey.toString()
+        const lak2fak = accessKeys.filter(({ public_key, access_key }) => 
+            public_key !== localPublicKeyStr &&
             access_key && access_key.permission && access_key.permission.FunctionCall &&
             access_key.permission.FunctionCall.receiver_id === accountId &&
             access_key.permission.FunctionCall.method_names &&
             access_key.permission.FunctionCall.method_names.length === 4 &&
             access_key.permission.FunctionCall.method_names.includes('add_request_and_confirm')    
         )
-        const confirmOnlyKey = PublicKey.from((await this.postSignedJson('/2fa/getAccessKey', { accountId })).publicKey)
+        // newLocalPublicKey and lak2fak -> full access keys, existing localStorageKey is deleted
         const actions = [
             deleteKey(confirmOnlyKey),
+            deleteKey(localPublicKey),
+            addKey(newLocalPublicKey, null),
             ...lak2fak.map(({ public_key }) => deleteKey(public_key)),
             ...lak2fak.map(({ public_key }) => addKey(public_key, null)),
             deployContract(contractBytes),
         ]
         console.log('disabling 2fa for', accountId)
-        return await this.signAndSendTransaction(accountId, actions)
+        let result
+        try {
+            result = await this.signAndSendTransaction(accountId, actions)
+        } catch (e) {
+            if (!/RetriesExceeded/.test(e.toString())) {
```

Comment:
nit: this is thrown right out of near-api-js, so it has `type` set, no need for regexps